### PR TITLE
03 1168: Add ability to cancel (`void`) active visit

### PR DESCRIPTION
**Pull request by donaldkibet**
## Requirements

- [x] This PR has a title that briefly describes the work done, including the ticket number if there is a ticket.
- [x] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide).
- [x] My work includes tests or is validated by existing tests.

## Summary
1. Add the ability to cancel the current active visit i.e the visit is voided. The reasoning behind this is feedback from clinical testing.
2. Fix bug on `start-visit-form` where if the time is 12 noon, it converts the time to the next day date. e.g `15-march-2022 12:03 PM` converted to `16-march-2022 00:03 AM'

## Screenshots
Kapture 2022-03-15 at 15 34 42

## Related Issue
https://issues.openmrs.org/projects/O3/issues/O3-1168?filter=myopenissues

### CODE REVIEW

**Comment by nanfuka** [comment]
@donaldkibet won't you need to write tests for the added code?

File: packages/esm-patient-chart-app/src/actions-buttons/cancel-visit.component.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  const handleClick = React.useCallback(
+    () =>
+      window.dispatchEvent(
+        new CustomEvent('visit-dialog', {
+          detail: {
+            type: 'cancel',
+          },
+        }),
+      ),
+    [currentVisit],
```
Comment (brandones) [inline packages/esm-patient-chart-app/src/actions-buttons/cancel-visit.component.tsx:22]:
```suggestion
    [],
```

per the linter

File: packages/esm-patient-chart-app/src/visit/visit-dialog.component.tsx [hunk 2]
```diff
@@ -1,24 +1,18 @@
...
-      <StartVisitPrompt isModalOpen={type == 'prompt'} closeModal={closeModal} state={state} />
-      <EndVisitPrompt isModalOpen={type === 'end'} patientUuid={patientUuid} closeModal={closeModal} />
-    </>
-  );
-};
+const VisitDialog: React.FC<VisitDialogProps> = ({ patientUuid }) => (
+  <>
+    <StartVisit patientUuid={patientUuid} />
+    <EndVisit patientUuid={patientUuid} />
+    <CancelVisit patientUuid={patientUuid} />
```
Comment (brandones) [inline packages/esm-patient-chart-app/src/visit/visit-dialog.component.tsx:14]: Two nits, totally optional:
1. This looks like 3 components stacked on top of one another. I think checking the `type` and only mounting the relevant one would be preferable in terms of code clarity.
2. I'd call each of those `StartVisitDialog`, `EndVisitDialog`, etc.

File: packages/esm-patient-chart-app/src/visit/visit-prompt/end-visit.test.tsx [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+const mockShowNotification = showNotification as jest.Mock;
+const mockMutate = jest.fn();
+
+jest.mock('@openmrs/esm-framework', () => {
+  const originalModule = jest.requireActual('@openmrs/esm-framework');
+  return {
+    ...originalModule,
+    showToast: jest.fn(),
+    showNotification: jest.fn(),
+    updateVisit: jest.fn(),
```
Comment (brandones) [inline packages/esm-patient-chart-app/src/visit/visit-prompt/end-visit.test.tsx:29]: Let's add mocks for these to core if they don't already exist

**Pull request merged**